### PR TITLE
add integration test for keywords function of user module

### LINKED ISSUE #276: Integration test for lib.analysis.user.keywords() function
The correct location for this test would be /test/integration-test/test_lib/test_analysis/test_user.py.

The integration test would patch the config dictionary, but would not patch any of the function calls.

### PR DESCRIPTION
**Pull request by Kausam**
## Integration tests for user module
Fix #276 . (Add Reference to the issue if any)

Changes proposed in this pull request:
- add integration test for keywords function of user.py
- put the common data files of unit and integration test in "test/data" directory

## Checks
- [ ] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Make sure you have added required documentation
- [ ] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@kaushik-rohit
@prasadtalasila

### CODE REVIEW

File: test/integration-test/test_lib/test_analysis/test_user.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import unittest
+from lib.analysis import user
+import lib.config as config
+from mock import patch
+import lib.util as util
+import os
+import StringIO
+import sys
+
+current_directory = os.path.dirname(os.path.realpath(__file__))
```
Comment (bhuvangupta008) [inline test/integration-test/test_lib/test_analysis/test_user.py]: global parameter. Please move it under setUp

**Review by bhuvangupta008 — CHANGES_REQUESTED** [review]

**Comment by bhuvangupta008** [comment]
No coverage report would be generated because the integration_test.sh is still empty.

File: test/integration-test/test_lib/test_analysis/test_user.py (hunk 1 above)
Comment (prasadtalasila) [reply test/integration-test/test_lib/test_analysis/test_user.py]: @Kausam The corresponding [unit test](https://github.com/prasadtalasila/IRCLogParser/blob/master/test/unit-test/test_lib/test_analysis/test_user.py) also has a global variable. Please update the unit test as well.

File: test/unit-test/test_lib/test_analysis/test_user.py [hunk 2]
```diff
@@ -16,9 +16,9 @@
 class UserTest(unittest.TestCase):
```
Comment (prasadtalasila) [inline test/unit-test/test_lib/test_analysis/test_user.py:15]: The variable on line-14 should move to setUp() function and assigned to null in tearDown() function.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

**Review by prasadtalasila — APPROVED** [review]

**Comment by prasadtalasila** [comment]
@Kausam and @bhuvangupta008 The integration tests are not running on travis. Please see [the log](https://travis-ci.org/prasadtalasila/IRCLogParser/jobs/344732579).

**Comment by prasadtalasila** [comment]
The reason is that the [test.sh](https://github.com/prasadtalasila/IRCLogParser/blob/master/test/integration-test/integration_test.sh) of integration tests is empty. It needs to have the following lines.
```
set -ex
coverage run --source=. -m unittest discover -s test/unit-test
coverage report -m
codeclimate-test-reporter --file .coverage --token $CODECLIMATE_REPO_TOKEN
```
Please add the `set -ex` command at the beginning of test.sh script in unit_test/ and functional_test/ directories.

**Comment by kaushik-rohit** [comment]
@Kausam The problem seems to be in test/integration-test/integration_test.sh at the line where you are using code-climate-test-reporter.

**Comment by bhuvangupta008** [comment]
the travis build failing problem seems to be the repo token not being set properly. the commit which i pushed built fine on my travis account but fails when building in the main repo

**Review by bhuvangupta008 — APPROVED** [review]

**Pull request merged**